[jenkins] values.yaml and casc: suggestion

hi jenkins chart guys,

just taking the latest chart for a spin `1.10.0` and i really like the CasC integration!

one behavior that i noticed is that if i place another key under `master.JCasC.configScript` that contains `jenkins.systemMessage`, like:
```
master:
  JCasC:
    configScripts:
      my-key: |
        jenkins:
          systemMessage: 'my system message'
```

it will clash and fail (see below) with the example in `values.yaml`:

```yaml
master:
  JCasC:
    configScripts:
      welcome-message: |
        jenkins:
          systemMessage: Welcome to our CI\CD server.  This Jenkins is configured and managed 'as code'.
```
__stack trace__:
```
SEVERE: Failed ConfigurationAsCode.init
java.lang.Error: java.lang.reflect.InvocationTargetException
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:110)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:175)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1096)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:104)
	... 8 more
Caused by: io.jenkins.plugins.casc.ConfiguratorException: Failed to read YamlSource: /var/jenkins_home/casc_configs/welcome-message.yaml
	at io.jenkins.plugins.casc.yaml.YamlUtils.merge(YamlUtils.java:42)
	at io.jenkins.plugins.casc.yaml.YamlUtils.loadFrom(YamlUtils.java:105)
	at io.jenkins.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:548)
	at io.jenkins.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:277)
	at io.jenkins.plugins.casc.ConfigurationAsCode.init(ConfigurationAsCode.java:269)
	... 13 more
Caused by: io.jenkins.plugins.casc.ConfiguratorException: Found conflicting configuration at YamlSource: /var/jenkins_home/casc_configs/welcome-message.yaml  in /var/jenkins_home/casc_configs/welcome-message.yaml, line 2, column 18:
      systemMessage: Welcome to our CI\CD server.  Th ... 
                     ^
	at io.jenkins.plugins.casc.yaml.YamlUtils.merge(YamlUtils.java:95)
	at io.jenkins.plugins.casc.yaml.YamlUtils.merge(YamlUtils.java:81)
	at io.jenkins.plugins.casc.yaml.YamlUtils.merge(YamlUtils.java:81)
	at io.jenkins.plugins.casc.yaml.YamlUtils.merge(YamlUtils.java:39)
	... 17 more
```
---

i can get around this by overriding `master.JCasC.configScripts.welcome-message` explicitly and not including it in any other `configScripts` key, but i'm guessing that isn't the intent of the default `values.yaml` file.

so just wanted to bring to your attention, among things to consider might include:

1. commenting out `welcome-message` in `values.yaml` or perhaps 
1. consider making `master.JCasC.configScripts` an array such that it will be overlaid completely instead of merged when doing a `helm upgrade`

regards,
tony

--- a/stable/jenkins/CHANGELOG.md
+++ b/stable/jenkins/CHANGELOG.md
@@ -6,6 +6,10 @@ numbering uses [semantic versioning](http://semver.org).
 
 NOTE: The change log until version 1.5.7 is auto generated based on git commits. Those include a reference to the git commit to be able to get more details.
 
+## 1.7.4
+
+Comments out JCasC example of jenkins.systemMessage so that it can be used by end users. Previously, an attempt to set systemMessage causes Jenkins to startup, citing duplicate JCasC settings for systemMessage [issue #13333](https://github.com/helm/charts/issues/13333)
+
 ## 1.7.2
 
 Update kubernetes-plugin to version 1.18.2 which fixes frequently encountered [JENKINS-59000](https://issues.jenkins-ci.org/plugins/servlet/mobile#issue/JENKINS-59000)

--- a/stable/jenkins/Chart.yaml
+++ b/stable/jenkins/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: jenkins
 home: https://jenkins.io/
-version: 1.7.3
+version: 1.7.4
 appVersion: lts
 description: Open source continuous integration server. It supports multiple SCM tools
   including CVS, Subversion and Git. It can execute Apache Ant and Apache Maven-based

--- a/stable/jenkins/values.yaml
+++ b/stable/jenkins/values.yaml
@@ -206,9 +206,9 @@ master:
     # configuration as code support plugin is no longer needed
     supportPluginVersion: "1.18"
     configScripts:
-      welcome-message: |
-        jenkins:
-          systemMessage: Welcome to our CI\CD server.  This Jenkins is configured and managed 'as code'.
+    # welcome-message: |
+    #   jenkins:
+    #     systemMessage: Welcome to our CI\CD server.  This Jenkins is configured and managed 'as code'.
 
   # Optionally specify additional init-containers
   customInitContainers: []

